packages/components/tests/integration/components/hds/dropdown/index-test.js

```diff
@@ -1,27 +1,100 @@
+  // NAMED YIELDS
+
+  test('it renders the "toggle" sub-components', async function (assert) {
+    assert.expect(2);
+    await render(hbs`
+      <Hds::Dropdown id="test-dropdown" as |dd|>
+        <dd.ToggleButton @text="toggle button" id="test-toggle-button" />
+        <dd.ToggleIcon @text="toggle icon" id="test-toggle-icon" />
+      </Hds::Dropdown>
+    `);
+    assert.dom('#test-dropdown #test-toggle-button').exists();
+    assert.dom('#test-dropdown #test-toggle-icon').exists();
+  });
+  test('it renders the "list-item" sub-components', async function (assert) {
+    assert.expect(6);
+    await render(hbs`
+      <Hds::Dropdown id="test-dropdown" as |dd|>
+        <dd.ToggleButton @text="toggle button" id="test-toggle-button" />
+        <dd.Description @text="description" id="test-list-item-description" />
+        <dd.Generic id="test-list-item-generic" />
+        <dd.Interactive @route="components.dropdown" @text="interactive" id="test-list-item-interactive" />
+        <dd.Separator id="test-list-item-separator" />
+        <dd.Title @text="title" id="test-list-item-title" />
+      </Hds::Dropdown>
+    `);
+    await click('button#test-toggle-button');
+    assert.dom('#test-dropdown ul').exists();
+    assert.dom('#test-dropdown #test-list-item-description').exists();
+    assert.dom('#test-dropdown #test-list-item-generic').exists();
+    assert.dom('#test-dropdown #test-list-item-interactive').exists();
+    assert.dom('#test-dropdown #test-list-item-separator').exists();
+    assert.dom('#test-dropdown #test-list-item-title').exists();
+  });
+
+  // POSITION
 
-  // TODO once everything is finalized
-  // - test @toggle prop or yielded toggles?
-  // - test yielded list items? (use single render with multiple assertions)
-  // - test @listPosition
+  test('it should render the content aligned on the right by default', async function (assert) {
+    await render(hbs`
+      <Hds::Dropdown id="test-dropdown" as |dd|>
+        <dd.ToggleButton @text="toggle button" id="test-toggle-button" />
+        <dd.Interactive @route="components.dropdown" @text="interactive" />
+      </Hds::Dropdown>
+    `);
+    await click('button#test-toggle-button');
+    assert
+      .dom('#test-dropdown ul')
+      .hasClass('hds-dropdown-list--position-right');
+  });
+  test('it should render the content aligned on the left if the value of @listPosition is "left"', async function (assert) {
+    await render(hbs`
+      <Hds::Dropdown id="test-dropdown" @listPosition="left" as |dd|>
+        <dd.ToggleButton @text="toggle button" id="test-toggle-button" />
+        <dd.Interactive @route="components.dropdown" @text="interactive" />
+      </Hds::Dropdown>
+    `);
+    await click('button#test-toggle-button');
+    assert
+      .dom('#test-dropdown ul')
+      .hasClass('hds-dropdown-list--position-left');
+  });
+
+  // WIDTH
+
+  test('it should render the content with a fixed width if a @width value is passed', async function (assert) {
```

Comment:
[question] what should happen if no width is defined?

Reply:
nothing, it just has min/max width. but this part of the styling, is not an integration test (will be tested by Percy)